### PR TITLE
Display log in container

### PR DESCRIPTION
**Pull request by Ennowulff**
so far it is not possible to display a log in a container.
added method CONTAINER to interface ZIF_LOGGER and implementation to ZCL_LOGGER.

```ABAP
REPORT zlogger_demo_07.

PARAMETERS p_info  TYPE c LENGTH 70 DEFAULT TEXT-001 MODIF ID dsp. "PRESS ENTER TO DISPLAY LOG
PARAMETERS p_info2 TYPE c LENGTH 70 DEFAULT TEXT-002 MODIF ID dsp. "PRESS ENTER AGAIN TO REFRESH LOG

INITIALIZATION.
  "create docking container
  DATA docker TYPE REF TO cl_gui_docking_container.
  CREATE OBJECT docker
    EXPORTING
      no_autodef_progid_dynnr = abap_true
      ratio                   = 80
      side                    = cl_gui_docking_container=>dock_at_bottom.
  "create display profile
  DATA my_profile TYPE REF TO zif_logger_display_profile.
  my_profile = zcl_logger_factory=>create_display_profile(
                 i_single_log  = abap_true )->set_grid( abap_true ).

  "Create logger
  DATA my_logger TYPE REF TO zif_logger.
  my_logger = zcl_logger_factory=>create_log( desc = 'ABAP Logger Demo 07' ) ##no_text.

AT SELECTION-SCREEN OUTPUT.
  "hide parameters (for information only)
  LOOP AT SCREEN.
    IF screen-group1 = 'DSP'.
      screen-input = '0'.
      MODIFY SCREEN.
    ENDIF.
  ENDLOOP.

AT SELECTION-SCREEN.

  "put some messages
  my_logger->w( obj_to_log = 'demonstration warning' ).
  my_logger->e( obj_to_log = 'demonstration error' ).
  my_logger->s( obj_to_log = 'demonstration success' ).

  "Display messages in container
  my_logger->container(
    i_container       = docker
    i_display_profile = my_profile->get( ) ).

START-OF-SELECTION.
```

### CODE REVIEW

**Comment by larshp** [comment]
@Ennowulff small trick, see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks for how to get syntax highlighting in the code block comments

File: src/zcl_logger.clas.abap [hunk 1]
```diff
@@ -727,4 +728,45 @@ CLASS zcl_logger IMPLEMENTATION.
...
+      "create control to display data
+      CALL FUNCTION 'BAL_CNTL_CREATE'
+        EXPORTING
+          i_container         = i_container
+          i_s_display_profile = display_profile
+          i_t_log_handle      = log_handles
+        IMPORTING
+          e_control_handle    = me->control_handle
+        EXCEPTIONS
+          OTHERS              = 1.
```
Comment (christianguenter2) [inline src/zcl_logger.clas.abap:762]: If these never fails what about adding
```ABAP
ASSERT sy-subrc = 0.
```
?

File: src/zcl_logger.clas.abap [hunk 2]
```diff
@@ -727,4 +728,45 @@ CLASS zcl_logger IMPLEMENTATION.
...
+    ELSE.
+      "refresh control
+      CALL FUNCTION 'BAL_CNTL_REFRESH'
+        EXPORTING
+          i_control_handle = me->control_handle
+          i_t_log_handle   = log_handles.
```
Comment (christianguenter2) [inline src/zcl_logger.clas.abap:768]: There are the exceptions 
```ABAP
  EXCEPTIONS
    CONTROL_NOT_FOUND
    INTERNAL_ERROR.
```
in my system. What about catching them?

**Review by christianguenter2 — CHANGES_REQUESTED** [review]
Two small remarks, otherwise we are good to go 👍

**Comment by christianguenter2** [comment]
Can you add your demo report here?
https://github.com/ABAP-Logger/demos

**Comment by Ennowulff** [comment]
> Can you add your demo report here? https://github.com/ABAP-Logger/demos

I will after the ür is done. otherwise there will be an error in the demo-repo because of the missing method.

**Comment by christianguenter2** [comment]
> > Can you add your demo report here? https://github.com/ABAP-Logger/demos
> 
> I will after the ür is done. otherwise there will be an error in the demo-repo because of the missing method.

You could create a pr in the demo repo and link both together? Of course it should be merged only after this is merged.

**Review by christianguenter2 — APPROVED** [review]
:+1

**Comment by AlexandreHT** [comment]
Thanks for the PR.
Quick RFC before merging: this would be the third method for displaying the log. Wouldn't it be better if they were prefixed by display_ (display_fullscreen, display_popup, display_in_container)?
That would required aliases for backward compatibility for some time but we could start fresh for this method.

**Comment by mbtools** [comment]
We should merge #145 before resolving the conflict here. Otherwise LGTM.

**Review by mbtools — APPROVED** [review]
Works well. Demo in https://github.com/ABAP-Logger/demos/pull/6

Thanks, @Ennowulff !

PS: Gut Ding will Weile haben 😉

**Comment by AlexandreHT** [comment]
Thanks

**Pull request merged**